src/Microsoft.Health.Dicom.Core/Features/Store/DicomDatasetValidator.cs

```diff
@@ -90,22 +93,66 @@ string EnsureRequiredTagIsPresent(DicomTag dicomTag)
             }
             else
             {
-                ValidateIndexedItems(dicomDataset);
+                ValidateIndexedItems(dicomDataset, customTagEntries);
             }
         }
 
-        private void ValidateIndexedItems(DicomDataset dicomDataset)
+        private void ValidateIndexedItems(DicomDataset dicomDataset, IList<CustomTagEntry> customTagEntries)
         {
-            HashSet<DicomTag> indexableTags = QueryLimit.AllInstancesTags;
+            HashSet<DicomTag> standardTags = new HashSet<DicomTag>();
+            Dictionary<string, CustomTagEntry> privateTags = new Dictionary<string, CustomTagEntry>();
+            foreach (var customTagEntry in customTagEntries)
+            {
+                DicomTag dicomTag = DicomTag.Parse(customTagEntry.Path);
```

Comment:
Validate should not be doing this. Have knowledge of what is private or not. Its job is just to validate the value against the VR type. The models passed should have enough information for validators to do this

Reply:
The reason for seperating standard and private tag is that  we need retrieve associated DicomElement in different way.
Standard custom tag can be retrieved as: DicomDataset.GetDicomItem, while Private custom tag need to loop entire dataset and locate with tagPath, since we don't have privateCreator.